File: TF_Study_PythonCode/8/8.1.1.py
import tensorflow as tf
import numpy as np

# import tensorflow.examples.tutorials.mnist.input_data as input_data
from tensorflow.examples.tutorials.mnist import input_data

mnist = input_data.read_data_sets("../data/MNIST_data/", one_hot=True)

print('训练集 train 数量：', mnist.train.num_examples,
    ', 验证集 validation 数量：', mnist.validation.num_examples,
    ', 测试集 test 数量：', mnist.test.num_examples)

print('train images shape: ', mnist.train.images.shape, 
        'labels shape: ', mnist.train.labels.shape)

image0 = mnist.train.images[0]

print('imageLen= ', len(image0))
print('image.shape =', image0.shape)
print('image data=', image0)
print('image.reshape =', image0.reshape(28,28))

import matplotlib.pyplot as plt
def plot_image(image):
    plt.imshow(image.reshape(28,28), cmap='binary')
    plt.show()

plot_image(mnist.train.images[1])

File: MOOC_TF_Study/_8/8_1_1.cs
using System;
using System.Collections.Generic;
using System.Text;
using MOOC_TF_Study.Common;
using NumSharp;
using Tensorflow;
using Tensorflow.Hub;
using static Tensorflow.Binding;

namespace MOOC_TF_Study._8
{
    class _8_1_1
    {
        public static void Run()
        {
            fun1();
        }

        private static void fun1()
        {
            var path = "../../../../data/MNIST_data/";
            // 这里需要引用 Tensorflow.Hub 项目
            var mnist = MnistModelLoader.LoadAsync(path, oneHot:true).Result;

            Console.WriteLine($"训练集 train 数量:{mnist.Train.NumOfExamples}");
            Console.WriteLine($"验证集 validation 数量:{mnist.Validation.NumOfExamples}");
            Console.WriteLine($"测试集 test 数量:{mnist.Test.NumOfExamples}");

            Console.WriteLine($"train images shape: {mnist.Train.Data.Shape}");
            Console.WriteLine($"labels shape:{ mnist.Train.Labels.Shape}");

            var image = mnist.Train.Data[0];

            Console.WriteLine($"image shape={image.Shape}");
            Console.WriteLine($"image data={image.ToString()}");
            Console.WriteLine($"image.reshape={image.reshape(28,28)}");

            // 画图是不可能画图了的。

        }

    }
}
